### DISCUSSION TITLE
Array of MC_MoveAbsoluteCA

### DISCUSSION THREAD

**Discussion by benn1993** [Beckhoff-USA-Community/XTS_Base#28] [General]
Hi,

thanks for the great project. I have one question about Mover (FB).

Why are the fbMoveAbsCA, fbGearInPosCA, fbMoveRelCA arrays? And they all refer to the same axis:

fbMoveAbsCA[0]( Axis := AxisReference );
fbMoveAbsCA[1]( Axis := AxisReference );

Why you use two axis for one mover? Shouldn't MC_MoveAbsoluteCA be used for a single axis and therefore one mover (https://infosys.beckhoff.com/english.php?content=../content/1033/tf5410_tc3_collision_avoidance/1419730315.html&id=6949934259970075206)?

Anyway, thanks for the great repo and i am hoping you answer me back.

**Comment by pswift066** [comment]
Hi, 

Thanks for checking out the project, and the great question. 

The underlying motion function blocks are not able to interrupt themselves with a secondary rising edge of Execute with updated parameters. See "BufferMode" section here: 

https://infosys.beckhoff.com/english.php?content=../content/1033/tcplclibmc2/458355339.html&id=

By having two instances of each motion function block, the two blocks of MC_MoveAbsoluteCA can alternate, taking turns interrupting each other with BufferMode MC_ABORTING.

This means that if you were to call Mover.MoveToPosition(x);, and then call Mover.MoveToPosition(x+50); before the first command had completed, the Mover would respond as you'd expect: abort the first move,  and immediately adjust course to the second destination while in motion.

Peter

**Reply by benn1993** [reply]
Thanks a lot for your answer!

That is a very smart move from you. Great project, thank you so much.
